tests/test_middlewares.py

```diff
@@ -368,3 +376,23 @@ def test_base_path_is_removed_before_routing(self):
         self.assertEqual(self.srmock.status, falcon.HTTP_200)
         self.simulate_request('/base_pathIncorrect/sub_path')
         self.assertEqual(self.srmock.status, falcon.HTTP_404)
+
+
+class TestResourceMiddleware(TestMiddleware):
+
+    def test_can_access_resource_params(self):
+        """Test that params can be accessed from within process_resource"""
+        global context
+
+        class Resource:
+            def on_get(self, req, resp, **params):
+                resp.body = params
```

Comment:
Nitpick: The fact that this works without serializing to a string is an implementation detail that may or may not hold up in the future. Instead, you may want to pass `params` through `json.dumps()`, assigning the result to `resp.data`.